core/condition_test.go

```diff
@@ -287,6 +287,48 @@ func TestCondition_evaluate(t *testing.T) {
 			ExpectedSuccess: true,
 			ExpectedOutput:  "[BODY].data.id > 0",
 		},
+		{
+			Name:            "body-jsonpath-hexadecimal-int-using-greater-than",
+			Condition:       Condition("[BODY].data > 0"),
+			Result:          &Result{Body: []byte("{\"data\": \"0x1\"}")},
```

Comment:
In hindsight, shouldn't the data here be `0x1` instead of `"0x1"`?

Could you add a test with the same body, but the condition `[BODY].data == 0x1`?

Reply:
I tried that, than it crashed.

JSON does not support hex numbers directly, we need to wrap them as strings. See this discussion: https://stackoverflow.com/questions/52671719/can-hex-format-be-used-with-json-files-if-so-how

We can only have decimal integers in a JSON element.

Reply:
That's alright. Could you still create a test to see if the condition `[BODY].data == 0x1` would work with the the body `[]byte("{\"data\": \"0x1\"}")`?

Reply:
It works!

I also added other tests of equality as well. Let me know what you think.

Reply:
I think you're missing my point. I'd like to make sure if the condition `[BODY].data == 0x1` passes if the body is `{"data": "0x1"}`.

All the tests you made have conditions that are numerical while the body is non-numerical (e.g. hexadecimal), but there are no tests with, for instance, an hexadecimal condition.

The test should look something like this:
```go
{
	Name:            "...",
	Condition:       Condition("[BODY].data == 0x1"),
	Result:          &Result{Body: []byte("{\"data\": \"0x1\"}")},
	ExpectedSuccess: true,
	ExpectedOutput:  "[BODY].data == 0x1",
},
```

Reply:
Btw, I added a bunch of tests, if you want I can remove some of them. Let me know what you think!

Reply:
Awesome, thank you!